crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -809,4 +1017,30 @@ mod tests {
             })
         );
     }
+
+    #[test]
+    pub fn coco_detection_dataset() {
+        let dataset = ImageFolderDataset::new_coco_detection(COCO_JSON, COCO_IMAGES).unwrap();
+        assert_eq!(dataset.len(), 3); // we have only three images defined
+        assert_eq!(dataset.get(3), None);
+
+        for item in dataset.iter() {
+            let file_name = Path::new(&item.image_path).file_name().unwrap();
+            match item.annotation {
+                // check if the number of bounding boxes is correct
+                Annotation::BoundingBoxes(v) => {
+                    if file_name == "two_dots_and_triangle.jpg" {
+                        assert_eq!(v.len(), 3);
+                    } else if file_name == "dot_triangle.jpg" {
+                        assert_eq!(v.len(), 2);
+                    } else if file_name == "one_dot.jpg" {
+                        assert_eq!(v.len(), 1);
+                    } else {
+                        panic!("{}", format!("unexpected image name: {}", item.image_path));
+                    }
```

Comment:
I would add a check to validate the expected bounding boxes, not just the number of bounding boxes.

Reply:
I was not sure if having all those coordinates in the test wasn't too much, but sure, I can add the checks for the actual boxes too.

Reply:
I think it's valuable to make sure we are parsing the correct values.